includes/class-wc-stripe-blocks-support.php

```diff
@@ -100,19 +100,55 @@ public function get_payment_method_data() {
 			$this->get_payment_request_javascript_params(),
 			// Blocks-specific options
 			[
-				'title'          => $this->get_title(),
-				'icons'          => $this->get_icons(),
-				'supports'       => $this->get_supported_features(),
-				'showSavedCards' => $this->get_show_saved_cards(),
-				'showSaveOption' => $this->get_show_save_option(),
-				'isAdmin'        => is_admin(),
-				'button'         => [
+				'title'                          => $this->get_title(),
+				'icons'                          => $this->get_icons(),
+				'supports'                       => $this->get_supported_features(),
+				'showSavedCards'                 => $this->get_show_saved_cards(),
+				'showSaveOption'                 => $this->get_show_save_option(),
+				'isAdmin'                        => is_admin(),
+				'shouldShowPaymentRequestButton' => $this->should_show_payment_request_button(),
+				'button'                         => [
 					'customLabel' => $this->payment_request_configuration->get_button_label(),
 				],
 			]
 		);
 	}
 
+	/**
+	 * Returns true if the PRB should be shown on the current page, false otherwise.
+	 *
+	 * Note: We use `has_block()` in this function, which isn't supported until WP 5.0. However,
+	 * WooCommerce Blocks hasn't supported a WP version lower than 5.0 since 2019. Since this
+	 * function is only called when the WooCommerce Blocks extension is available, it should be
+	 * safe to call `has_block()` here.
+	 * That said, we only run those checks if the `has_block()` function exists, just in case.
+	 *
+	 * @return boolean  True if PRBs should be displayed, false otherwise
+	 */
+	private function should_show_payment_request_button() {
+		// TODO: Remove the `function_exists()` check once the minimum WP version has been bumpted
+		//       to version 5.0.
+		if ( function_exists( 'has_block' ) ) {
+			// Don't show if PRBs are supposed to be hidden on the cart page.
+			if (
+				has_block( 'woocommerce/cart' )
+				&& ! apply_filters( 'wc_stripe_show_payment_request_on_cart', true )
+			) {
+				return false;
+			}
```

Comment:
Fixed in e2b566ae335893ba78ce3f8b272d532603d5d381.

I made the filter show the PRB on checkout by default _for the checkout block_ because I know that's how the Blocks team wanted the Block to work — it's also how the bundled Stripe integration works in the Blocks plugin.

Let me know if you think we should stick to the same functionality used for the shortcode flow (show on cart, hide on checkout).

Reply:
I'm ok with enabling it to keep the current behavior for blocks, but there's an outstanding issue #1586 which should overlap with this. Basically we will copy the same approach from WCPay and have a setting to display it at specific locations (product, cart and checkout), and remove the filter. This should hide or display the PRB for both checkout block and shortcode. For consistency.

So I don't know what's best to do, TBH. Let me know what you think.

Reply:
Let's go with the current implementation for now, just to be consistent with what our blocks integration is replacing 🤔 

We can be more consistent when the configuration in #1586 enables a more user friendly we to control this 🙂